contracts/registry/BaseRegistry.sol

```diff
@@ -1,101 +1,91 @@
 pragma solidity ^0.4.11;
 
 // This is the base contract that your contract PocketNodeRegistry extends from.
-contract BaseRegistry {
+contract BaseRegistry is NodeCrud {
 
     // The owner of this registry.
     address public owner = msg.sender;
 
     uint public creationTime = now;
 
-    // This struct keeps all data for a Record.
-    struct Record {
-        // Keeps the address of this record creator.
-        address owner;
-        // Keeps the time when this record was created.
-        uint time;
-        // Keeps the index of the keys array for fast lookup
-        uint keysIndex;
-        string url;
-    }
+    // List of registered Nodes
+    address[] public registeredNodes;
 
     // This mapping keeps the records of this Registry.
-    mapping(address => Record) records;
-
-    // Keeps the total numbers of records in this Registry.
-    uint public numRecords;
+    mapping(address => Node) nodeRecords;
 
-    // Keeps a list of all keys to interate the records.
-    address[] public keys;
+    // Keeps the total numbers of Node records in this Registry.
+    uint public numNodeRecords;
 
+    // Keeps a list of all keys to iterate the Node records.
+    address[] public nodeKeys;
 
     modifier onlyOwner {
         if (msg.sender != owner) revert();
         _;
     }
 
-
-
     // This is the function that actually insert a record.
-    function register(address key, string url) {
-        if (records[key].time == 0) {
-            records[key].time = now;
-            records[key].owner = msg.sender;
-            records[key].keysIndex = keys.length;
-            keys.length++;
-            keys[keys.length - 1] = key;
-            records[key].url = url;
-            numRecords++;
+    function registerNodeRecord(address key, string[] _supportedTokens, string _url, uint8 _port, uint _index) {
```

Comment:
Don't call the first parameter `key`, call it `_nodeAddress`, also shouldn't we pass in here wheter or not this node is an oracle or a relayer?